### PR TITLE
Create a ML Pipelines page in Programming Guide (issue 1263)

### LINKED ISSUE #1263: Create a ML Pipelines page in Programming Guide
We should create a ML Pipelines page in Programming Guide that describes how to use BigDL (for training and inference) in Spark ML pipelines; it should combine the DLEstimate and DLClassifer pages, but also add DLModel & DLClassifierModel

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?

Create a ML Pipelines page in Programming Guide by merging the DLClassifier and DLEstimator doc.

resolve: https://github.com/intel-analytics/BigDL/issues/1263

## How was this patch tested?

doc

### CODE REVIEW

**Comment by hhbyyh** [comment]
cc @jason-dai @shane-huang @wzhongyuan for the doc review. Thanks

File: docs/docs/getting-started.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ BigDL supports loading models trained and saved in BigDL, or a trained Tensorflo
 
 Once you have a loaded model, you can call `Module.predict()` to do predictions. Note that you need to convert your input data into proper format which `predict` accepts. For how to prepare your data, refer to section [Prepare your Data](#prepare-your-data). 
 
-If you are using predicted result as a component inside a Spark ML pipeline, refer to [DLEsitimator API](ProgrammingGuide/MLPipeline/DLEstimator.md) and [DLClassifier API](ProgrammingGuide/MLPipeline/DLClassifier.md) for usage. 
+If you are using predicted result as a component inside a Spark ML pipeline, refer to [DLEstimator API](ProgrammingGuide/MLPipeline/MLPipeline.md) and [DLClassifier API](ProgrammingGuide/MLPipeline/MLPipeline.md) for usage.
```
Comment (jason-dai) [inline docs/docs/getting-started.md]: Refer to the Using Spark ML Pipeline page

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -31,9 +31,8 @@ pages:
     - Using Sequential API: ProgrammingGuide/Model/Sequential.md
     - Using Functional API: ProgrammingGuide/Model/Functional.md
   - Optimization: ProgrammingGuide/optimization.md
-  - Use MLPipeline: 
-    - DLEstimator: ProgrammingGuide/MLPipeline/DLEstimator.md
-    - DLClassifier: ProgrammingGuide/MLPipeline/DLClassifier.md
+  - Use MLPipeline:
```
Comment (jason-dai) [inline docs/mkdocs.yml]: Change to:
Use Spark ML: ProgrammingGuide/MLPipeline.md

File: docs/docs/ProgrammingGuide/MLPipeline/MLPipeline.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
+
+## Use BigDL with Apache Spark ML pipeline ##
+
+BigDL provides `DLEstimator` and `DLClassifier` for users with Apache Spark MLlib experience, which
+provides high level API for training a BigDL Model with the Apache Spark `Estimator`/`Transfomer`
+pattern, thus users can conveniently fit BigDL into a ML pipeline.
+
```
Comment (jason-dai) [inline docs/docs/ProgrammingGuide/MLPipeline/MLPipeline.md]: Add short description of DLModel and DLClassifierModel, esp. for the user to directly load a pre-trained model and use it in Spark ML for prediction.

File: docs/docs/ProgrammingGuide/MLPipeline/MLPipeline.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+Currently only scala interface are implemented for `DLEstimator` and `DLClassifier`. Python
+support will be added soon.
+
+
+## DLEstimator ##
+
+`DLEstimator` extends `org.apache.spark.ml.Estimator` and supports model training from
+Apache Spark DataFrame/Dataset. 
+ 
+Different from many algorithms in MLlib, `DLEstimator` supports more data types for the
```
Comment (jason-dai) [inline docs/docs/ProgrammingGuide/MLPipeline/MLPipeline.md]: in Spark MLlib

**Comment by hhbyyh** [comment]
Thanks for the review @jason-dai . Updated according to comments and merge the whole example code into the document since the we need special SparkContext initialization.

**Comment by hhbyyh** [comment]
PR validation passed: http://172.168.2.101:8080/job/BigDL-PR-Validation/1379/

**Comment by hhbyyh** [comment]
merge it first to catch the release, let me know if further change is required.

**Pull request merged**